### PR TITLE
Enable tide, test trigger, deck and greenhouse

### PR DESCRIPTION
**Pull request by rmohr**
Mostly finalize the prow setup. As an example a test job is configured for `rmohr/kubevirt`.

Add the following components:

 * `deck` for viewing prow jobs and build logs
  - entrypoint: https://deck-kubevirt-prow.apps.ovirt.org
  - example build result: https://deck-kubevirt-prow.apps.ovirt.org/view/gcs/kubevirt/pr-logs/pull/rmohr_kubevirt/15/pull-kubevirt-test/1094735307942662144
  - dashboard: https://deck-kubevirt-prow.apps.ovirt.org/pr?query=is%3Apr%20state%3Aopen%20author%3Armohr
  - example of a merged PR: https://deck-kubevirt-prow.apps.ovirt.org/pr?query=is%3Apr+repo%3Armohr%2Fkubevirt+author%3Armohr+head%3Armohr-patch-6
 * `trigger` plugin for  triggering jobs
 * `override` plugin, to allow overriding required jobs, in case a merge needs to happen anyway.
 * `tide` for for merging and pool-based presubmit tests

If adopted for `kubevirt/kubevirt` this allows us to avoid broken master branches by merging different PRs at the same time, since tide will create a merge pool if possible, or at least test them sequentially before merging. 

As part of this PR, it enables the tide workflow for this repo, including a prow-config check.

### CODE REVIEW

**Comment by slintes** [comment]
So this means we are heading to the lgtm / approve / automatic merge flow? I like that. How will `trigger` and  `tide` interact with Travis and STDCI?
`deck` looks nice, never saw that before :+1:

File: github/ci/prow/files/config.yaml [hunk 1]
```diff
@@ -1,7 +1,61 @@
...
   max_prowjob_age: 48h
   max_pod_age: 30m
 
 prowjob_namespace: kubevirt-prow
 pod_namespace: kubevirt-prow
+log_level: debug
+
+tide:
+  merge_method:
+    rmohr/kubevirt: squash
```
[the remark this answers is not in the document]
Comment (rmohr) [reply github/ci/prow/files/config.yaml]: At the end yes. I just want the current state to be merged but still not use it on kubevirt/kubevirt before we have all people on board.

File: github/ci/prow/files/job-config.yaml [hunk 2]
```diff
@@ -91,3 +91,20 @@ periodics:
     - name: token
       secret:
         secretName: oauth-token
+
+presubmits:
```
[the remark this answers is not in the document]
Comment (rmohr) [reply github/ci/prow/files/job-config.yaml]: It does not hurt and is nice for playing until we agree on using it for kubevirt/kubevirt. It is right now deployed.

**Comment by rmohr** [comment]
@cynepco3hahue @slintes removed references to rmohr/kubevirt and instead use this repo as an example for the prow workflow. The `check-prow-config` job runs `checkconfig` from prow to check if our prow configs are valid. As soon as one of use types the `/lgtm`, tide  will merge the PR. The PR is already live.

**Comment by rmohr** [comment]
> Great job, I just unsure how will we merge it with std-ci?

So, tide  can be configured to require all extgernal test contexts to pass, or whitelist only expect some specific lanes to pass.

**Comment by rmohr** [comment]
@slintes yeah I am an approver in the OWNERS file. Therefor it got autoapproved. That can be changed with a configuration value. So the only thing which  is missing with the current configuration is a `/lgtm` (which the tide status says).

The `/lgtm` is configured to not be allowed to come from the PR creator. Let me write it here:

/lgtm

kubevirt-bot should complain with a comment:

**Comment by slintes** [comment]
ah ok, got it. Auto-Approve is fine for me.

**Comment by rmohr** [comment]
@slintes as discussed offline, made it now explicitly necessary to always add a `/approve`. Note that to `lgtm` a PR, you can either type `/lgtm`/, or do a normal github review and click `approve` via the usual github mechanism.

**Comment by slintes** [comment]
> you can either type `/lgtm`, or do a normal github review

too bad this doesn't work the other way arround: after my `/lgtm` it still asks for my review on top of this page. (but I was assigned... whatever that means on a PR)

**Comment by rmohr** [comment]
> too bad this doesn't work the other way arround: after my `/lgtm` it still asks for my review on top of this page. (but I was assigned... whatever that means on a PR)

Yeah. You were assigned because you are listed as a reviewer in https://github.com/kubevirt/project-infra/blob/master/OWNERS

**Comment by slintes** [comment]
do we a have definition of what an assignee of a PR is doing? Seems to be unclear....
https://stackoverflow.com/questions/41087206/on-github-whats-the-difference-between-reviewer-and-assignee
(not relevant for this PR, just wondering)

**Comment by rmohr** [comment]
@slintes my comment regarding to the "assigned" state was wrong. You would have been added as a "reviewer" because of the OWNERS file. You got assigned because of the "lgtm". I guess "assigned" has project specific meaning. In our case, i think, it is there to allows you to query for all PRs easily where you ever added lgtm or approved, so that you can find PRs easier where some interactions from you are still required.

**Pull request merged**